src/components/TaskList/tasksList.js

```diff
@@ -0,0 +1,66 @@
+import React from 'react';
+import { useState } from 'react';
+import DeleteTaskButton from '../DeleteTask/deleteTask.js';
+import '/Users/mergestackemployee/todo/src/components/TaskList/tasklist.css';
+import EditTaskButton from '../EditTask/editTask.js';
+import Checkbox from '/Users/mergestackemployee/todo/src/components/CheckBox/checkBox.js';
+
+const TaskList = ({ tasks, setTasks }) => {
+    const [editableTaskId, setEditableTaskId] = useState(null);
+
+    const handleToggleEdit = (taskId) => {
+        setEditableTaskId(taskId === editableTaskId ? null : taskId);
+    };
+
+    const handleSaveTask = (taskId, newName) => {
+        setTasks((prevTasks) =>
+            prevTasks.map((task) =>
+                task.id === taskId ? { ...task, name: newName } : task
+            )
+        );
+        handleToggleEdit(taskId);
+    };
+
+    const handleToggleComplete = (taskId) => {
```

Comment:
There is a duplication `handleToggleComplete` is defined in CheckBox. So we don't this here.